src/cartridges/int_adyen_SFRA/cartridge/controllers/middlewares/confirm.js

```diff
@@ -0,0 +1,49 @@
+const OrderMgr = require('dw/order/OrderMgr');
+const adyenGetOriginKey = require('*/cartridge/scripts/adyenGetOriginKey');
+const AdyenHelper = require('*/cartridge/scripts/util/adyenHelper');
+
+function confirm(req, res, next) {
+  const order = OrderMgr.getOrder(req.querystring.ID);
+  const paymentMethod = order.custom.Adyen_paymentMethod;
+
+  if (
+    AdyenHelper.getAdyenGivingEnabled() &&
+    AdyenHelper.isAdyenGivingAvailable(paymentMethod)
+  ) {
+    const protocol = req.https ? 'https' : 'http';
+    const originKey = adyenGetOriginKey.getOriginKeyFromRequest(
+      protocol,
+      req.host,
+    );
+    const environment = AdyenHelper.getAdyenEnvironment().toLowerCase();
+    const configuredAmounts = AdyenHelper.getDonationAmounts();
+    const charityName = AdyenHelper.getAdyenGivingCharityName();
+    const charityWebsite = AdyenHelper.getAdyenGivingCharityWebsite();
+    const charityDescription = AdyenHelper.getAdyenGivingCharityDescription();
+    const adyenGivingBackgroundUrl = AdyenHelper.getAdyenGivingBackgroundUrl();
+    const adyenGivingLogoUrl = AdyenHelper.getAdyenGivingLogoUrl();
+
+    const donationAmounts = {
+      currency: session.currency.currencyCode,
+      values: configuredAmounts,
+    };
+
+    const viewData = res.getViewData();
+    viewData.adyen = {
+      originKey,
+      environment,
+      adyenGivingAvailable: true,
```

Comment:
We might want to look into this to check whether it is available and not set it always to true